src/components/AddTransaction/index.js

```diff
@@ -1,26 +1,50 @@
-import {useState} from 'react';
+import {useEffect, useState} from 'react';
 import {useParams} from 'react-router-dom';
 
 import useStore from '../../hooks/useStore';
 import {Form} from '../AddNewAccount/styled';
 
-export default function AddTranscation() {
+export default function AddTransaction() {
 	const [transaction, setTransaction] = useState({
 		date: '',
 		change: '',
 		note: '',
 		category: '',
 	});
 
-	const {accountID} = useParams();
+	const {accountID, transactionID} = useParams();
+	const currentBankAccount = useStore(state =>
+		state.db.find(account => account.id === accountID)
+	);
+
+	const currentTransaction = currentBankAccount.entries.find(entry => entry.id === transactionID);
+
 	const addTransaction = useStore(state => state.addTransaction);
+	const editTransaction = useStore(state => state.editTransaction);
+
+	useEffect(() => {
+		if (currentTransaction) {
+			setTransaction({
+				id: currentTransaction.id,
+				date: currentTransaction.date,
+				change: currentTransaction.change,
+				note: currentTransaction.note,
+				category: currentTransaction.category,
+			});
+		}
+	}, [currentTransaction]);
 
 	return (
 		<>
 			<Form
 				onSubmit={event => {
 					event.preventDefault();
-					addTransaction(accountID, transaction);
+					if (transactionID) {
+						editTransaction(accountID, transaction);
+						alert('edited');
```

Comment:
I would probably not work with an `alert`. If you want notifications, you could have a look at [React Toastify](https://www.npmjs.com/package/react-toastify)

Reply:
We thought so too and we want to implement toastify in a seperate user story to make alerts happen whenever the user edits, adds or deletes something!